### PR TITLE
Add Topology LP copyright to all source files

### PR DESCRIPTION
**Pull request by davidchappelle**
Since Topology has made significant contributions to
the entire project we have decided to explicitly
include them in the copyright.

### CODE REVIEW

**Comment by jpetso** [comment]
I was going to comment on the other thread but didn't get around to finish my comment.

I don't agree with the "determine copyright from commit history" argument, which I think is misguided because:
* People will copy files out of the repo (e.g. as part of installation) without git history, and copyright information should be preserved through that process.
* Git seems pretty stable and popular now, but then so did Subversion in its heyday. Countless commit histories have been lost because they weren't deemed worthwhile enough to spend time and effort on.
* Determining copyright from commit histories is a royal pain, because it basically involves an audit of lots of commits to see which ones count as copyrightable and which ones don't. Making sure that this information is part of the latest version of the file saves interested parties a lot of time.

I feel like it's a bad trade-off to accept those drawbacks to skip on the small amount of effort that it takes to add/maintain copyright years.

Apart from that, I see other issues with the proposed line format:
* It divides contributors into two classes - "important" ones listed by name, and "less important" ones only referred to anonymously.
* It doesn't obsolete any commit history analysis work, which is sort of the entire point of listing copyright in the file at all. If you don't list all copyright holders individually, you might as well make the line read `Copyright (c) whoever committed to this file, refer to git commit history`.
* If one or two more named contributors are added, the line will get unreasonably long, so it's not a very extensible format.

I suggest to use a format used by many seasoned open source projects:
```
// Copyright (c) 2015 Tavendo GmbH
// Copyright (c) 2015, 2016 Topology LP
// Copyright (c) 2016 Joe Schmoe <schmoe@example.com>
```

**Comment by oberstet** [comment]
@jpetso No, I disagree. Many projects are getting rid of dates. Also: no to listing all contributors too. Anyway: I won't merge that.

In that case, lets leave it like it is.

**Comment by oberstet** [comment]
> It divides contributors into two classes

Yes, I agree.

And since I don't want to get into such discussions and waste time, lets put it like this: I created the original work. An already fully working version. All rights to the code I wrote are with Tavendo. In the meantime, there have been contributions smaller and larger. Since we didn't require a contributor license assignment or agreement for contributors, these contributors essentially have copyrights in the respective portions of code (modulo the "inventive / creative amount" involved .. it's not about the technical number of lines changes .. eg replacing \n with \r\n in all files wouldn't result in a copyright in all these lines). Topology certainly has contributors the most code of all contributors, and certainly also with respective inventive/creative height. But as you say: where is the bar for adding a contributor in explicit form to the headers?

In summary, I think a more correct (and still sustainable) notice would be

```
Copyright (c) Tavendo GmbH and contributors.
```

@davidchappelle IOW: I am not going to merge the PR, as the discussion above already supports my concern here https://github.com/crossbario/autobahn-cpp/pull/93#issuecomment-184838316

> I am cautious, because I want to avoid being asked to do similar for X and Y and Z on all Autobahn's - I wouldn't!

**Comment by oberstet** [comment]
@jpetso Ok, fair enough!

If you do care about this, consider opening a new issue asking to change the notice to `Copyright (c) Tavendo GmbH and contributors.`. I'd then asked around other devs, as I think _if_ we do that change (which I'd be +1 on), then we probably should do it for all Autobahn's. Because the situation is basically the same. Original authorship me (copyrights with Tavendo) plus lots of contributions (and hence implicit copyrights with contributors for respective portions).

On the other hand, even if we do _not_ do that change, the legal situation would be the same: the copyrights of contributors exist, even without notice. The notice just makes that more visible, discoverable, obvious. As long as a contributor can proof his/her contributions, the copyright would be enforceable even without any notice at all. And the proof is the Git history.

There is even more "implicit stuff" going on: when a contributor opens a PR, it's taken as granted that he/she accepts that the contributions are licensed under the Boost license - the very same as the code being merged into.

> Sidenote: in the US, it is possible to _register_ a copyright by filing source code at the USPTO. Doing so will allow to claim higher damages in case of infringement and some other details I forgot. This is a tedious thing though: Alex has been in conversation with them, since you have to file the whole source code, and if you do change it, you have to file again. Filing just a Git repo SHA1 isn't acceptable for them=( They love the red tape;)
